lang/wasm_alien_thread_runner.hh

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (C) 2023-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+#pragma once
+
+#include <functional>
+#include <mutex>
+#include <queue>
+#include <condition_variable>
+#include <thread>
+
+#include <seastar/core/future.hh>
+#include <seastar/util/noncopyable_function.hh>
+
+#include "rust/cxx.h"
+#include "rust/wasmtime_bindings.hh"
+
+namespace wasm {
+
+struct wasm_compile_task {
+    seastar::noncopyable_function<void()> func;
+    seastar::promise<rust::Box<wasmtime::Module>>& done;
+    unsigned shard;
+};
+
+struct task_queue {
+    std::mutex _mut;
+    std::condition_variable _cv;
+    static constexpr unsigned QUEUE_SIZE = 128;
```

Comment:
it seems we don't use this variable anymore after switching to unbounded queue and using the combination of `mutex` and `condition_variable` for coordinating the consumer and producer of the queue.

Reply:
btw, an unbounded queue has its own obvious problems. But I think it cannot grow in an unbounded way because Raft will serialize schema changes.